packages/@aws-cdk/aws-elasticloadbalancing/lib/load-balancer.ts

```diff
@@ -266,6 +274,7 @@ export class LoadBalancer extends Resource implements IConnectable {
   public addListener(listener: LoadBalancerListener): ListenerPort {
     const protocol = ifUndefinedLazy(listener.externalProtocol, () => wellKnownProtocol(listener.externalPort));
     const instancePort = listener.internalPort || listener.externalPort;
+    const sslCertificateArn = listener.sslCertificateId || listener.sslCertificateArn;
```

Comment:
I'd prefer we default to using the newer property over the old one.

Bonus points for creating some error handling in the case where a user specifies both.

```suggestion
    const sslCertificateArn = listener.sslCertificateArn || listener.sslCertificateId;
```

Reply:
@njlynch Thanks for the comments. When I was writing a unit test to test error handling, there was an error "Property 'toThrowError' does not exist on type 'StackInspector'". I also tried to add @aws-cdk/assert/jest, rebuilt etc, but the error persisted. Any idea why this is? Thanks.

Reply:
Depends. What did the test look like? Typically, it would look something like this:

```ts
expect(() => lb.addListener(...)).toThrow(/error message here/);
```

Reply:
So, in the load-balancer.ts I had
<pre><code>if (listener.sslCertificateArn && listener.sslCertificateId) {
      throw new Error('"sslCertificateId" is deprecated, please use "sslCertificateArn" only.');
    }</code></pre>

The relevant part of test was like you said:

<pre><code>expect(() =>
    lb.addListener({
      externalPort: 80,
      internalPort: 8080,
      sslCertificateArn: sslCertificateArn,
    })).toThrow(/"sslCertificateId" is deprecated, please use "sslCertificateArn" only./);</code></pre>

Reply:
Ah, yes, I see now.

The `expect` is currently coming from `@aws-cdk/assert`, which is overriding the default jest expect. This is a holdover from the older nodeunit-style tests here.

The following should work:
1. Import `@aws-cdk/assert/jest` (as you said you tried).
2. Remove the imports from `'@aws-cdk/assert'`.
3. Rewrite the existing expects from `expect(stack).to(haveResource(....));` to `expect(stack).toHaveResource(...);`.

Can you give that a shot?